File: src/tests/DataBlockASPNETDemo/vb.net/BusinessFacade/BookPersistentObject.vb
Imports System
Imports System.Data
Imports voidsoft.DataBlock
Namespace BookManager.BusinessFacade

	Public Class BookPersistentObject
	Inherits voidsoft.DataBlock.PersistentObject

		Public Sub New(ByVal database As DatabaseServer, ByVal connectionString As String, ByVal mainTable As TableMetadata)
			MyBase.New(database, connectionString, mainTable)
		End Sub

		Public Sub New(ByVal session As Session, ByVal mainTable As TableMetadata)
			MyBase.New(session, mainTable)
		End Sub

		Public Sub New(ByVal mainTable As TableMetadata)
			MyBase.New(mainTable)
		End Sub
	End Class 
End Namespace

File: src/tests/DataBlockASPNETDemo/csharp/BusinessFacade/BookPersistentObject.cs
using System;
using System.Data;
using voidsoft.DataBlock;
namespace BookManager.BusinessFacade
{
    public class BookPersistentObject : voidsoft.DataBlock.PersistentObject
    {

        public BookPersistentObject(DatabaseServer database, string connectionString, TableMetadata mainTable)
            : base(database, connectionString, mainTable)
        {
        }

        public BookPersistentObject(Session session, TableMetadata mainTable)
            : base(session, mainTable)
        {
        }
        public BookPersistentObject(TableMetadata mainTable)
            : base(mainTable)
        {
        }

    }
}
